lib/src/main/java/com/orange/ods/compose/component/button/OdsButton.kt

```diff
@@ -0,0 +1,99 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonColors
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.ButtonElevation
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.Icon
+import androidx.compose.material.IconToggleButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.Typography
+import androidx.compose.material.ripple.LocalRippleTheme
+import androidx.compose.material.ripple.RippleTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.orange.ods.compose.theme.DarkSurfaceDefault
+import com.orange.ods.compose.theme.Grey800
+import com.orange.ods.compose.theme.OdsPrimaryRippleTheme
+import androidx.compose.runtime.getValue
+import com.orange.ods.compose.theme.OdsNoRippleTheme
+
+/**
+ * Ripple theme used on primary color.
+ * Note: It is only used on contained buttons.
+ */
+private object OdsOnPrimaryRippleTheme : RippleTheme {
+    @Composable
+    override fun defaultColor()= RippleTheme.defaultRippleColor(
+        contentColor = MaterialTheme.colors.onPrimary,
+        lightTheme = true // allow to force ripple used on primary color when in dark mode
+    )
+
+    @Composable
+    override fun rippleAlpha() = RippleTheme.defaultRippleAlpha(
+        MaterialTheme.colors.primary,
+        lightTheme = !isSystemInDarkTheme()
+    )
+}
+
+@Composable
+fun OdsButton(
+    text: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    @DrawableRes
+    iconRes: Int? = null,
+    enabled: Boolean = true,
+    interactionSource: MutableInteractionSource = remember { MutableInteractionSource() }
+) {
+    CompositionLocalProvider(LocalRippleTheme provides OdsOnPrimaryRippleTheme) {
+        Button(
+            onClick = onClick,
+            enabled = enabled,
+            modifier = modifier,
+            interactionSource = interactionSource,
+            elevation = null,
+            shape = odsButtonShape,
+            colors = ButtonDefaults.buttonColors(
+                disabledBackgroundColor = if (isSystemInDarkTheme()) Grey800 else MaterialTheme.colors.primary.copy(alpha = ContentAlpha.disabled),
+                disabledContentColor = if (isSystemInDarkTheme()) DarkSurfaceDefault else MaterialTheme.colors.onSurface.copy(alpha = ContentAlpha.disabled)
+            )
+        ) {
+            iconRes?.let { ButtonIcon(it) }
```

Comment:
on a pas de gestion left and right pour l'icone dans le design system, c'est forcement à gauche?

Reply:
hum. Très bonne question ça m'étonnerait en fait. Pour moi en arabe l'icône devrait être à droite. Je sens le truc où on va nous demander de passer en RTL à la volée aussi comme on a fait pour le light/dark